### ISSUE TITLE
failed: ClusterShardingFailureSpec

### ISSUE THREAD

**Issue by seglo** [akka/akka#28676]
https://jenkins.akka.io:8498/job/akka-artery-cluster-tests/2275/consoleFull

```
[info] [JVM-2] - must recover after journal/network failure (on node 'first', class akka.cluster.sharding.DDataClusterShardingFailureMultiJvmNode2) *** FAILED ***
[info] [JVM-2]   java.lang.AssertionError: assertion failed: received unexpected message RealMessage(Value(40,4),Actor[akka://MultiNodeClusterShardingSpec/system/sharding/Entity/4/40#-1910323477]) after 100 millis
[info] [JVM-2]   at akka.testkit.TestKitBase.expectNoMsg_internal(TestKit.scala:727)
[info] [JVM-2]   at akka.testkit.TestKitBase.expectNoMessage(TestKit.scala:691)
[info] [JVM-2]   at akka.testkit.TestKitBase.expectNoMessage$(TestKit.scala:690)
[info] [JVM-2]   at akka.testkit.TestKit.expectNoMessage(TestKit.scala:929)
[info] [JVM-2]   at akka.cluster.sharding.ClusterShardingFailureSpec.$anonfun$new$8(ClusterShardingFailureSpec.scala:143)
[info] [JVM-2]   at akka.remote.testkit.MultiNodeSpec.runOn(MultiNodeSpec.scala:384)
[info] [JVM-2]   at akka.cluster.sharding.ClusterShardingFailureSpec.$anonfun$new$6(ClusterShardingFailureSpec.scala:138)
[info] [JVM-2]   at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
[info] [JVM-2]   at akka.testkit.TestKitBase.within(TestKit.scala:376)
[info] [JVM-2]   at akka.testkit.TestKitBase.within$(TestKit.scala:364)
```

```
[info] [JVM-1] [ERROR] [02/28/2020 10:06:00.482] [MultiNodeClusterShardingSpec-akka.actor.default-dispatcher-22] [akka://MultiNodeClusterShardingSpec/system/controller/barriers] unannounced disconnect of RoleName(first) (akka.remote.testconductor.BarrierCoordinator$ClientLost: unannounced disconnect of RoleName(first))
[info] [JVM-1] - must recover after journal/network failure (on node 'controller', class akka.cluster.sharding.DDataClusterShardingFailureMultiJvmNode1) *** FAILED ***
[info] [JVM-1]   java.lang.RuntimeException: barrier failed: first-delayed
[info] [JVM-1]   at akka.remote.testconductor.ClientFSM$$anonfun$3.applyOrElse(Player.scala:241)
[info] [JVM-1]   at akka.remote.testconductor.ClientFSM$$anonfun$3.applyOrElse(Player.scala:215)
[info] [JVM-1]   at scala.runtime.AbstractPartialFunction.apply(AbstractPartialFunction.scala:38)
[info] [JVM-1]   at akka.actor.FSM.processEvent(FSM.scala:801)
[info] [JVM-1]   at akka.actor.FSM.processEvent$(FSM.scala:798)
[info] [JVM-1]   at akka.remote.testconductor.ClientFSM.akka$actor$LoggingFSM$$super$processEvent(Player.scala:168)
[info] [JVM-1]   at akka.actor.LoggingFSM.processEvent(FSM.scala:941)
[info] [JVM-1]   at akka.actor.LoggingFSM.processEvent$(FSM.scala:923)
[info] [JVM-1]   at akka.remote.testconductor.ClientFSM.processEvent(Player.scala:168)
[info] [JVM-1]   at akka.actor.FSM.akka$actor$FSM$$processMsg(FSM.scala:795)
[info] [JVM-1]   ...
```

```
[info] [JVM-3] - must recover after journal/network failure (on node 'second', class akka.cluster.sharding.DDataClusterShardingFailureMultiJvmNode3) *** FAILED ***
[info] [JVM-3]   java.lang.RuntimeException: barrier failed: first-delayed
[info] [JVM-3]   at akka.remote.testconductor.ClientFSM$$anonfun$3.applyOrElse(Player.scala:241)
[info] [JVM-3]   at akka.remote.testconductor.ClientFSM$$anonfun$3.applyOrElse(Player.scala:215)
[info] [JVM-3]   at scala.runtime.AbstractPartialFunction.apply(AbstractPartialFunction.scala:38)
[info] [JVM-3]   at akka.actor.FSM.processEvent(FSM.scala:801)
[info] [JVM-3]   at akka.actor.FSM.processEvent$(FSM.scala:798)
[info] [JVM-3]   at akka.remote.testconductor.ClientFSM.akka$actor$LoggingFSM$$super$processEvent(Player.scala:168)
[info] [JVM-3]   at akka.actor.LoggingFSM.processEvent(FSM.scala:941)
[info] [JVM-3]   at akka.actor.LoggingFSM.processEvent$(FSM.scala:923)
[info] [JVM-3]   at akka.remote.testconductor.ClientFSM.processEvent(Player.scala:168)
[info] [JVM-3]   at akka.actor.FSM.akka$actor$FSM$$processMsg(FSM.scala:795)
[info] [JVM-3]   ...
```

Maybe related closed failed issues:
* #26724
* #24013

**Comment by seglo** [comment]
Also on akka-multi-node-nightly build

https://jenkins.akka.io:8498/job/akka-multi-node-nightly/7626/consoleFull

**Comment by seglo** [comment]
https://jenkins.akka.io:8498/job/akka-nightly-jdk8/7429/consoleFull

**Comment by johanandren** [comment]
I think this is the same as fixed by #28671
